### PR TITLE
drivers: adc: Add threshold_count & threshold_reg_offset to NPCX adc config structure

### PR DESCRIPTION
**Pull request by perezpri**

### CODE REVIEW

**Comment by perezpri** [comment]
@keith-zephyr @MaureenHelm

File: drivers/adc/adc_npcx.c [hunk 1]
```diff
@@ -52,6 +49,8 @@ struct adc_npcx_config {
 	struct npcx_clk_cfg clk_cfg;
 	/* pinmux configuration */
 	const struct npcx_alt *alts_list;
+	/* amount of thresholds supported */
+	uint8_t threshold_count;
```
Comment (ChiHuaL) [inline drivers/adc/adc_npcx.c:55]: As the threshold_count is fixed, how about add the **const** qualifer?

Comment (perezpri) [reply drivers/adc/adc_npcx.c]: Done.

**Review by MaureenHelm — DISMISSED** [review]

**Review by ChiHuaL — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by perezpri** [comment]
> Hey change looks ok but what's the motivation? If I'm reading it right there's no change in behavior, so is this just for the symmetry of having dt properties in the config structure? It'd be nice if the patch comment or PR comment clarifies that.
> 
> See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-body

Done, I have updated commit messages and mentioned that driver behavior is not impacted. Thanks.

**Comment by fabiobaltieri** [comment]
> > Hey change looks ok but what's the motivation? If I'm reading it right there's no change in behavior, so is this just for the symmetry of having dt properties in the config structure? It'd be nice if the patch comment or PR comment clarifies that.
> > See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-body
> 
> Done, I have updated commit messages and mentioned that driver behavior is not impacted. Thanks.

Thanks that makes it easier to review... still though, why the change? Sorry not meaning to be pedantic about it, genuinely curious.

**Review by fabiobaltieri — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**